TutorLizard.BusinessLogic/Services/StudentService.cs

```diff
@@ -1,11 +1,41 @@
-﻿using TutorLizard.BusinessLogic.Interfaces.Services;
+﻿using Microsoft.EntityFrameworkCore;
+using TutorLizard.BusinessLogic.Interfaces.Data.Repositories;
+using TutorLizard.BusinessLogic.Interfaces.Services;
+using TutorLizard.BusinessLogic.Models;
+using TutorLizard.BusinessLogic.Models.DTOs;
+using TutorLizard.BusinessLogic.Models.DTOs.Requests;
+using TutorLizard.BusinessLogic.Models.DTOs.Responses;
 
 namespace TutorLizard.BusinessLogic.Services;
 public class StudentService : IStudentService
 {
-    // TODO inject needed repositories
-    public StudentService()
+    private readonly IDbRepository<Ad> _adRepository;
+    private readonly IDbRepository<AdRequest> _adRequestRepository;
+    public StudentService(IDbRepository<Ad> adRepository,
+                          IDbRepository<AdRequest> adRequestRepository)
     {
-        
+        _adRepository = adRepository;
+        _adRequestRepository = adRequestRepository;
+    }
+
+    public async Task<StudentsAcceptedAdsResponse> ViewAcceptedAds(StudentsAcceptedAdsRequest request)
+    {
+        var studentId = request.StudentId;
+
+        var acceptedAdRequests = await _adRequestRepository.GetAll()
```

Comment:
If you want later to access the navigation properties, you have to use `.Include()`, otherwise, each `AdRequest` you get in this call will have `Ad = null`. You will not only need `Ad`, but also `Category` and `User` in said `Ad`, so I suggest:
```
            .Include(ar => ar.Ad)
            .ThenInclude(ad => ad.Category)
            .Include(ar => ar.Ad)
            .ThenInclude(ad => ad.User)
```
[Docs](https://learn.microsoft.com/en-us/ef/core/querying/related-data/eager#including-multiple-levels) tell me that it's ok to call multiple includes on the same property.

> You may want to include multiple related entities for one of the entities that is being included. For example, when querying Blogs, you include Posts and then want to include both the Author and Tags of the Posts. To include both, you need to specify each include path starting at the root. For example, Blog -> Posts -> Author and Blog -> Posts -> Tags. It doesn't mean you'll get redundant joins; in most cases, EF will combine the joins when generating SQL.

Reply:
Thanks. Finished.